src/main/java/emissary/kff/KffFile.java

```diff
@@ -143,42 +138,34 @@ public String getPreferredAlgorithm() {
      * @param crc Result of CRC calculation
      * @return true if the record is in the list, false if it isn't
      */
-    private boolean binaryFileSearch(@Nullable byte[] hash, long crc) {
-        if (hash == null && myPreferredAlgorithm != null) {
```

Comment:
The only call to this method is `check` which already checks hash for `null`. Since this is a private method, I'm thinking we can remove this useless check and add `@Nonnull` to the parameter declaration.